sdk/program/src/instruction.rs

```diff
@@ -207,6 +214,19 @@ impl Instruction {
             accounts,
         }
     }
+
+    pub fn new_with_borsh<T: BorshSerialize>(
```

Comment:
Might be helpful to deprecate `Instruction::new` since it has the hidden bincode serialization and instead add `new_with_vec` and `new_with_bincode`?

Reply:
Finally the right terms! Is `new_with_vec` meant for just `Vec<u8>` input?

Reply:
Yeah, `new_with_vec` takes a `Vec<u8>` on the assumption that it is in its final form.  Possibly already serialized by some other developer-defined mechanism.